### PR TITLE
vttest: prep for go queryservice test

### PR DESCRIPTION
**Pull request by sougou**
@alainjobart 
This has the changes needed to launch a standalone mysql instance.

### CODE REVIEW

File: py/vttest/environment.py [hunk 1]
```diff
@@ -28,7 +28,14 @@
 
 def get_test_directory():
   """Returns the toplevel directory for the tests. Might create it."""
-  directory = tempfile.mkdtemp(prefix='vt')
+  dataroot = os.environ['VTDATAROOT']
+  if dataroot:
+    directory = os.path.join(dataroot, "test")
```
Comment (alainjobart) [inline py/vttest/environment.py]: Could we make a tmp dir in there?

File: py/vttest/local_database.py [hunk 2]
```diff
@@ -50,9 +54,14 @@ def vtgate_addr(self):
 
   def config(self):
     """Returns a dict with enough information to be able to connect."""
-    return {
-        'port': vt_processes.vtgate_process.port,
-        }
+    if self.mysql_only:
+      return {
+          'socket': self.mysql_db.unix_socket(),
```
Comment (alainjobart) [inline py/vttest/local_database.py]: this is not going to work in google3, I think you need to let the mysql_db return whatever it wants, and just export it here. You also probably want to export the username, password, and whatever else makes sense.

File: py/vttest/environment.py (hunk 1 above)
Comment (sougou) [inline py/vttest/environment.py]: That was my first thought. But then, I felt that 'test' was more explicit. Ideally, generating a unique name would be awesome, but I couldn't find a good utility for it.
Or, how about 'local'?

File: py/vttest/mysql_flavor.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
+"""Define abstractions for various mysql flavors."""
```
Comment (alainjobart) [inline py/vttest/mysql_flavor.py]: please add a comment that this is used by mysql_db_mysqlctl.py only.

**Comment by alainjobart** [comment]
Couple details but looks good overall!

File: py/vttest/local_database.py (hunk 2 above)
Comment (sougou) [inline py/vttest/local_database.py]: Yeah. My plan is to try importing this, and fixing forward on the necessary changes. This is just to get the initial test working.

File: py/vttest/environment.py (hunk 1 above)
Comment (alainjobart) [inline py/vttest/environment.py]: tempfile.mkdtemp takes a prefix, see below, but also a base dir (dir=...) so you could use that?
I think all these lines 31 to 38 could be just:
  directory = tempfile.mkdtemp(prefix='vttest', dir=os.environ.get('VTDATAROOT', None))

File: py/vttest/local_database.py (hunk 2 above)
Comment (alainjobart) [inline py/vttest/local_database.py]: OK cool. Go ahead and merge this if you want to move forward...

(my feeling is you'll need mysql_db to have a config_json() or something method that returns the dict, so the mysqlctl one can return exactly what you have there, and the google3 can return a port. Both should return also a username and password. Then your unit tests can just read these values and pass them to the db conn layer)

File: py/vttest/environment.py (hunk 1 above)
Comment (sougou) [inline py/vttest/environment.py]: Why didn't I think of that! That just works. I'll add the additional comment on the mysql_flavor & submit.

**Pull request merged**